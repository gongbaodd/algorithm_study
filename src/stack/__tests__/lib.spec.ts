import { Stack } from "../lib";

describe("Stack", () => {
  it("should work as a stack", () => {
    const stack = new Stack();

    expect(stack).toMatchSnapshot();
    expect(stack.isEmpty()).toEqual(true);
    expect(stack.size()).toEqual(0);

    stack.push(50);
    expect(stack).toMatchSnapshot();
    expect(stack.isEmpty()).toEqual(false);
    expect(stack.size()).toEqual(1);

    expect(stack.top()).toEqual(50);
    expect(stack).toMatchSnapshot();
  });
});
