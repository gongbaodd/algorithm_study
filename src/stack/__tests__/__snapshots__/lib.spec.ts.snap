// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack should work as a stack 1`] = `
Stack {
  "pHead": LNode {
    "data": null,
    "next": null,
  },
}
`;

exports[`Stack should work as a stack 2`] = `
Stack {
  "pHead": LNode {
    "data": null,
    "next": LNode {
      "data": 50,
      "next": null,
    },
  },
}
`;

exports[`Stack should work as a stack 3`] = `
Stack {
  "pHead": LNode {
    "data": null,
    "next": LNode {
      "data": 50,
      "next": null,
    },
  },
}
`;
